a10_octavia/controller/worker/tasks/vthunder_tasks.py

```diff
@@ -1254,3 +1257,41 @@ def execute(self, loadbalancer, vthunder_config, device_config_dict, flavor_data
                     raise exceptions.FlavorDeviceNotFound(device_flavor)
 
         return vthunder_config, False
+
+
+class GetListenersStats(VThunderBaseTask):
+    """Task for retrieving listener stats from vthunder"""
+
+    @axapi_client_decorator
+    def execute(self, vthunder):
+        try:
+            listener_stats = []
+            lb = self.loadbalancer_repo.get(db_apis.get_session(), id=vthunder.loadbalancer_id)
+            if lb.provisioning_status == constants.ACTIVE:
+                response = self.axapi_client.slb.virtual_server.stats(
+                    name=vthunder.loadbalancer_id,
+                    timeout=CONF.a10_health_manager.stats_update_timeout,
+                    max_retries=1)
+                if response:
+                    for stats in response['port-list']:
+                        listener = self.listener_repo.get(
+                            db_apis.get_session(),
+                            load_balancer_id=vthunder.loadbalancer_id,
+                            protocol_port=stats['port-number'])
```

Comment:
please help also add filter for protocol here. 
It is possible to have different protocol with same port number like:
```
slb virtual-server vip1 1.1.1.1
   port 80 tcp
      ...
   port 80 http
      ....
```

Reply:
As discussed on teams,

Octavia didnt allow me to create two listeners with same port number in a loadbalancer

stack@victoria:~$ openstack loadbalancer list
+--------------------------------------+------+----------------------------------+-------------+---------------------+------------------+----------+
| id                                   | name | project_id                       | vip_address | provisioning_status | operating_status | provider |
+--------------------------------------+------+----------------------------------+-------------+---------------------+------------------+----------+
| a30cff39-2da9-4927-a24d-3b067efef146 | v1   | bc83270643c14317b1e2653854c1e190 | 10.0.0.17   | ACTIVE              | OFFLINE          | a10      |
+--------------------------------------+------+----------------------------------+-------------+---------------------+------------------+----------+
stack@victoria:~$ openstack loadbalancer listener list
+--------------------------------------+-----------------+-----------+----------------------------------+----------+---------------+----------------+
| id                                   | default_pool_id | name      | project_id                       | protocol | protocol_port | admin_state_up |
+--------------------------------------+-----------------+-----------+----------------------------------+----------+---------------+----------------+
| a14067c6-be7f-4f31-a3c8-96d10031750e | None            | listener1 | bc83270643c14317b1e2653854c1e190 | TCP      |          8080 | True           |
+--------------------------------------+-----------------+-----------+----------------------------------+----------+---------------+----------------+
stack@victoria:~$ openstack loadbalancer listener create --protocol HTTP --protocol-port 8080 v1 --name listener2
Another Listener on this Load Balancer is already using protocol TCP and protocol_port 8080 (HTTP 409) (Request-ID: req-8e60833e-48da-4727-84fa-60d60877ea53)
stack@victoria:~$